feat/landing-sixth-step

Mirved64 opened the pull request:
Close #18

Mirved64 committed:
feat: add castom breakpoints
BREAKING CHANGE: castom breakpoints not work

Mirved64 commented:
@Nelfimov 

**Скоуп**
кастомные брейкпоинты

**Описание**
Если использовать кастомные брейкпоинты, то пропс, к которому они применяются не работает.

**Мои действия**

Создал `breakpoints.ts` добавил в `index.ts` в `ui/theme`. 

Сделал на главной странице красный `Box` с `height={{standard: 500, wide: 600, ultra:700}}`. В `DOM` бокс не отображается, пишет `height="[object Object]"`.

Читал доку к `styled-system`, чат Академии. В чате Вы писали, что нужно указывать для кастомных брейкпоинтов значения в виде объекта. Но так работать не хочет.

**Вопрос**
Что я упустил при настройки и использованию кастомных брейкпоинтов?

ПР сделал, ветка актуальна.

oxiqod commented:
@Mirved64 Покажи пожалуйста файл ‘breakpoints.ts’ из темы, и то как ты применяешь брейкпоинты в ‘display’, непосредственно в компонентах

TorinAsakura commented:
@oksssvv Он же написал, что ветка актуальна

oxiqod commented:
@Mirved64 

в общем дело такое:

`small` как ты назвал, я так понимаю это мобильное устройство? 
- если так - то его брейкпоинт я не вводила, он априори менее обычного `Standard` / `FullHD` поэтому его можно не прописывать, да и версия для планшика в макете не предусматривается 

- т.е. у тебя условно идут такие точки останова - `[ '80em', '160em', '240em']`
где `80em - standard`, `160em - wide` и `240em - ultra`, а то что менее первого бейкпоинта - мобилка

- указывая значения в виде объекта всё будет работать, + мобильную версию прописывай символом `_` (можешь поиграться с другими вариантами - не пробовала)

попробуй завтра, если не получится - пиши, + сразу скажу, после изменения значений брейкпоинта в теме -  отображения по новым брейкпоинтам на странице не будет, даж если стр перезагрузить, нужно перезапускать локалхост

Mirved64 commented:
> @Mirved64
> 
> в общем дело такое:
> 
> `small` как ты назвал, я так понимаю это мобильное устройство?
> 
> * если так - то его брейкпоинт я не вводила, он априори менее обычного `Standard` / `FullHD` поэтому его можно не прописывать, да и версия для планшика в макете не предусматривается
> * т.е. у тебя условно идут такие точки останова - `[ '80em', '160em', '240em']`
>   где `80em - standard`, `160em - wide` и `240em - ultra`, а то что менее первого бейкпоинта - мобилка
> * указывая значения в виде объекта всё будет работать, + мобильную версию прописывай символом `_` (можешь поиграться с другими вариантами - не пробовала)
> 
> попробуй завтра, если не получится - пиши, + сразу скажу, после изменения значений брейкпоинта в теме - отображения по новым брейкпоинтам на странице не будет, даж если стр перезагрузить, нужно перезапускать локалхост

@oksssvv 

После рестарта локалхоста бокс отображается с высотой как прописано  в `ultra`, т.е. `700px`. А вот при изменении вьюпорта брейкпоинт не срабатывает и высота не меняется.

Пробовал брейкпоинты задавать в `em`, `px`, просто числом и перезапускал локалхост - результат отрицательный.
Пробовал высоту задавать в `em`, `px`, просто числом и перезапускал локалхост - результат отрицательный.

Mirved64 committed:
feat: add wideDesktop, TV viewports in useWindowWidth

oxiqod commented:
@Mirved64 
последний коммит - актуальное состояние? 
всё еще вижу там `small` в брейкпоинтах

Mirved64 commented:
@oksssvv 

актуальное
убирал `small`, так же результат отрицательный

oxiqod commented:
@Mirved64 Ну как оно актуальное, если в брейкпоинтах я вижу small? 

я ж писала в данном случае достаточно указать standard, wide & ultra  и разберись сначала с брейкпоинтами, а затем, зная что они работают правильно, будешь делать useWindowWidth

Mirved64 commented:
@oksssvv 

В ближайшее время внесу правки с учетом замечаний.

Mirved64 commented:
@oksssvv 

Твой файл брейкпоинтов сработал. Благодарю за оперативную помощь.

Только смущает что линт отключать нужно при деструктуризации массива (когда присваиваем ключам объекта `breakpoints` значения элементов массива). Но при разрешенных способах деструктуризации брейкпоинты ломаются.

Еще раз спасибо!

Mirved64 committed:
fix: rework breakpoints.ts

Mirved64 committed:
feat: add backround image files, add wide and ultra screen in navigation, drawer, social link components

Mirved64 committed:
Merge branch 'feat/landing-fifth-step' into feat/landing-sixth-step

Mirved64 committed:
feat: add navigation, hero sections and navbar in wide screen

Mirved64 committed:
feat: add courses section in wide screen

Mirved64 committed:
Merge branch 'feat/landing-fifth-step' into feat/landing-sixth-step

Signed-off-by: Mikhail Medvedev <99605124+Mirved64@users.noreply.github.com>

Mirved64 committed:
feat: add process section in wide screen

Mirved64 committed:
feat: add faq section in wide screen

Mirved64 committed:
Merge branch 'feat/landing-fifth-step' into feat/landing-sixth-step

Mirved64 committed:
feat: add hero in ultra screen

Mirved64 committed:
feat: add navigation section in ultra screen

Mirved64 committed:
Merge branch 'feat/landing-fifth-step' into feat/landing-sixth-step

# Conflicts:
#	landing/fragments/landing-courses/src/cards/cards.component.tsx
#	landing/fragments/landing-courses/src/courses.component.tsx
#	landing/fragments/landing-courses/src/sidebar/sidebar.component.tsx
#	landing/fragments/landing-faq/src/faq.component.tsx
#	landing/fragments/landing-navigation/src/drawer/drawer.component.tsx
#	landing/fragments/landing-navigation/src/item/item.component.tsx
#	landing/fragments/landing-navigation/src/navigation.component.tsx
#	landing/fragments/landing-process/src/process.component.tsx
#	landing/fragments/landing-process/src/slider/slider.component.tsx
#	landing/fragments/landing-steps/src/steps.component.tsx
#	landing/fragments/landing-technologies/src/technologies.component.tsx
#	landing/pages/index-page/src/index.page.tsx
#	ui/accordion/src/accordion.component.tsx
#	ui/card/src/card.component.tsx
#	ui/drawer/src/drawer.component.tsx
#	ui/navbar/src/navbar.component.tsx
#	ui/slider/src/slider.component.tsx
#	ui/utils/src/hooks/use-window-width.hook.ts

Mirved64 committed:
fix: resolve merge errors

Mirved64 committed:
feat: add faq section in ultra display

Mirved64 committed:
feat: add process section in ultra display

Mirved64 committed:
feat: add courses section in ultra display

Mirved64 committed:
refactor: change layout in navigation section

Mirved64 committed:
refactor: change layout in hero section

Mirved64 committed:
fix: change versions of some dependencies

Nelfimov committed:
chore: generate styles for button with 0.0.15

Nelfimov committed:
chore: add colors to different states

Mirved64 committed:
BREAKING CHANGE: all button are grey color

Mirved64 committed:
fix: change ui/button dependencies versions

Mirved64 committed:
chore: check background images, background position

Mirved64 committed:
chore: tune parallax and fix layout in courses section wide screen

Mirved64 committed:
chore: change placeholder font family, resolve typo in input size

Mirved64 committed:
chore: check background images, background position in library page

Mirved64 committed:
chore: fix a typo in color name

Mirved64 committed:
chore: ui/button with @atls-ui-parts/button 0.0.16

Mirved64 committed:
feat: add animation in first section in ultra viewport
BREAKING CHANGE: parallax in first section not work, links and button in navigation not work

Mirved64 committed:
perf: replace condition to display in first section

Mirved64 committed:
feat: add animation in first section on hd viewport

Mirved64 committed:
feat: add animation in first section on wide viewport

Mirved64 committed:
feat: add animation in first section on ultra viewport

Mirved64 committed:
chore: change duration of animation in first section

Mirved64 committed:
feat: add header fragment

Mirved64 committed:
feat: add header and headerIndex components

Mirved64 committed:
feat: add header to library page

Mirved64 committed:
refactor: divide library fragment by library mobile and library

Mirved64 committed:
refactor: add display props in navigation-mobile component

Mirved64 committed:
feat: add animation in first section of library page

Mirved64 committed:
feat: add cards in library page for wide and ultra viewports

Mirved64 committed:
chore: rename ui/preload to ui/animate

Mirved64 committed:
chore: make library page layout is responsive

Mirved64 committed:
BREAKING CHANGE: animation on click not work

Mirved64 commented:
@Nelfimov 

**Скоуп**
Анимация по нажатию

**Описание**
Реализовать анимацию `HEADER` в проекте

**Мои действия**
- сделал хук [useDimensions](https://www.reddit.com/r/reactjs/comments/b0oh20/usedimensions_a_react_hook_to_measure_dom_nodes/), чтобы получать `height`, `top`, `y` у секций сайта - не могу получить доступ к полям `height`, `top`, `y`, `IDE` пишет что нет таких полей, хотя я указал `type`;
- задал [controls](https://www.framer.com/motion/use-animation-controls/#dynamic-start), для контроля анимации (запуска по клику на ссылку в хедере) - не понимаю как прокинуть `controls` в хедер, чтобы по клику запустить анимацию;
- думаю как указать промежуточные фреймы в анимации (поднять секцию на .25, .5 и .75 от высоты первой секции), чтобы анимация работала и на `wide` и на `ultra` вьюпортах.

**Вопросы**
- Правильно ли я рассуждаю о том как реализовать анимацию?
- Как использовать `useDimensions` если уже задан `ref` домузлу?
- Как реализовать запуск анимации по клику? (Прокинуть `controls` и написать функцию)
- Правильно ли я понимаю значения анимации на промежуточных фреймах?

ПР сделал, ветка актуальна.

Nelfimov commented:
> * Правильно ли я рассуждаю о том как реализовать анимацию?

Отвечу на этот вопрос так чтобы остальные тоже не остались без ответа 😁

Тебе достаточно сверстать элемент, который в изначальном состоянии находится за вьюпортом вверху, то есть `top: -100%`.

Этому компоненту надо передавать буленовский пропс, по которому он будет определять - показываться/выезжать или прятаться/заезжать. Анимация сводится к простому перемещению по оси Y.

Буленовский пропс будет у тебя стейтом, который изменяется по нажатию кнопки. Соответственно ты можешь:
- либо обозначить этот стейт в общем родителе у кнопки и анимированного компонента
- либо создать контекст и из него получать пропс

Mirved64 commented:
> > * Правильно ли я рассуждаю о том как реализовать анимацию?
> 
> Отвечу на этот вопрос так чтобы остальные тоже не остались без ответа 😁
> 
> Тебе достаточно сверстать элемент, который в изначальном состоянии находится за вьюпортом вверху, то есть `top: -100%`.
> 
> Этому компоненту надо передавать буленовский пропс, по которому он будет определять - показываться/выезжать или прятаться/заезжать. Анимация сводится к простому перемещению по оси Y.
> 
> Буленовский пропс будет у тебя стейтом, который изменяется по нажатию кнопки. Соответственно ты можешь:
> 
> * либо обозначить этот стейт в общем родителе у кнопки и анимированного компонента
> * либо создать контекст и из него получать пропс

Я ошибся с названием анимации. 
Нужно анимацию `header-nav-1` реализовать

Nelfimov commented:
> > > * Правильно ли я рассуждаю о том как реализовать анимацию?
> > 
> > 
> > Отвечу на этот вопрос так чтобы остальные тоже не остались без ответа 😁
> > Тебе достаточно сверстать элемент, который в изначальном состоянии находится за вьюпортом вверху, то есть `top: -100%`.
> > Этому компоненту надо передавать буленовский пропс, по которому он будет определять - показываться/выезжать или прятаться/заезжать. Анимация сводится к простому перемещению по оси Y.
> > Буленовский пропс будет у тебя стейтом, который изменяется по нажатию кнопки. Соответственно ты можешь:
> > 
> > * либо обозначить этот стейт в общем родителе у кнопки и анимированного компонента
> > * либо создать контекст и из него получать пропс
> 
> Я ошибся с названием анимации. Нужно анимацию `header-nav-1` реализовать

@oksssvv призываю тебя

oxiqod commented:
@Mirved64 оч мало что поняла, ты написал реализовать анимацию хедера в проекте, какая именно анимация? (на сколько помню там с анимацией связанны 2 вещи, drawer и скролл до секции ), можно контекст в дизайне, в коде? 
краткое описание что для чего должно быть сделано и как выглядеть в итоге?
пока не понимаю даж зачем тебе кастомный хук делать

Mirved64 commented:
@oksssvv
В макете, в 6 степе есть фрейм `header-nav-1`.  

Вот её не получается реализовать.

Там при клике по ссылке `ACADEMY` когда в самом низу страницы проигрывает я анимация - секции поднимаются до определённой точки (у каждой секции своя) и спускаются. 

Вот для задания этих точек каждой секции пишу хук, он пока не работает. 

Как по клику сделать анимацию я разобрался.

oxiqod commented:
@Mirved64
правильно ли поняла, что при клике по `ACADEMY` в навигации, вьюпорт должен прокуртится по конца `Hero`?
всё остальное у тебя работает?

Mirved64 commented:
@oksssvv

Не уверен что правильно понял реализацию анимации. 
Не могу из хука получить высоту элемента. 
В хуке нужно указать реф на элемент - не разберусь как правильно сделать: хочу получать высоту секции `academy`, но на ней уже есть реф

Mirved64 committed:
fix: useDimensions hook is work, merge refs of sections
BREAKING CHANGE: animation on click not work

Mirved64 committed:
fix: animation on click work

Mirved64 committed:
chore: check layout, check duplicate properties

oxiqod commented on landing/fragments/landing-about/src/about.component.tsx:
```diff
@@ -1,133 +1,124 @@
+          </Box>
+
+          <Box>
+            <Text color='text.white' fontSize='small' lineHeight='huge'>
```
заметила что одинаковая часть верстки используется 7 раз, можно вынести `Box` и `Text` в отдельный компонент и далее переиспользовать

<details>

![Снимок экрана от 2023-09-23 20-58-18](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/a32ddb48-0d5f-4763-b74d-ebe6c13196ef)
</details>

oxiqod commented on landing/fragments/landing-courses/src/cards/cards.component.tsx:
```diff
@@ -123,3 +136,37 @@ export const CardsSwiper = ({ spaceBetween, slidesPerView, className }: SwiperPr
 )
+
+export const CardsMaterialsWide = () => {
+  const { isWideDesktop, isTV } = useWindowWidth()
```
`isTV`?)
нет, эт просто большая диагональ монитора, с большей плотностью пикселей)

oxiqod commented on landing/fragments/landing-courses/src/cards/cards.component.tsx:
```diff
@@ -123,3 +136,37 @@ export const CardsSwiper = ({ spaceBetween, slidesPerView, className }: SwiperPr
+      ))}
+    </>
+  )
+}
```
такс, почему здесь 4 экспортируемых компонента? давай дели ответственность файла на несколько директорий)

oxiqod commented on landing/fragments/landing-courses/src/info/info.component.tsx:
```diff
@@ -8,121 +8,181 @@ import { Text }             from '@ui/text'
+          color='text.primary'
+          fontSize={{ _: 'small', standard: 'ordinary', ultra: 'strong' }}
+          lineHeight='huge'
+        >
```
ситуация как и в `about`,  11 одинаковых кусков верстки, если переиспользовать `Box` и `Text` то выйдет  строк на 60 меньше кода в файле 

<details>

![Снимок экрана от 2023-09-23 21-20-11](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/9265e270-1139-43ea-ae36-3bf84ded1417)
</details>

oxiqod commented on landing/fragments/landing-header/src/header-index.component.tsx:
```diff
@@ -0,0 +1,275 @@
+
+import { HeaderIndexProps }       from './header.interfaces'
+
+export const HeaderIndex: FC<HeaderIndexProps> = ({
```
так дела не делаются компонент слишком раздутый, нужна хоть какая то модульность, как минимум вынеси `drawerCards` отдельно

oxiqod replied:
что б добраться до начала верстки нужно крутануть более 3х раз, у тебя там есть крупная функция с анимацией, ей можно сделать отдельный файл, и по смыслу правильно и код чище смотрится

oxiqod commented on landing/fragments/landing-header/src/header.component.tsx:
```diff
@@ -0,0 +1,183 @@
+    id: index,
+  }))
+
+  const drawerCards = cardsList.map((card, index, array) => {
```
тоже самое, они ж вроде одинаковые с тем `drawerCards` который в `header-index.component.tsx`, в любом случае его нужно вынести отдельно и применить тогда уже в двух компонентах

oxiqod commented on landing/fragments/landing-header/src/header.component.tsx:
```diff
@@ -0,0 +1,183 @@
+            <Layout flexBasis={40} />
+          </Row>
+
+          {visible ? (
```
`Condition`

oxiqod commented on landing/fragments/landing-header/src/header-index.component.tsx:
```diff
@@ -0,0 +1,275 @@
+            <Layout flexBasis={40} />
+          </Row>
+
+          {visible ? (
```
`Condition`

oxiqod commented on landing/fragments/landing-hero/src/hero-wide.component.tsx:
зачем нужен отдельный компонент для широкого экрана? все можно реализовать в `hero.component.tsx`,  к чему дублировать верстку в данном случае?

oxiqod commented on landing/fragments/landing-hero/src/social-link/social-link.component.tsx:
```diff
@@ -36,3 +36,18 @@ export const SocialLinkMobile: FC<SocialLinkProps> = ({ icon, href }) => (
   </Box>
 )
+
+export const SocialLinkUltra: FC<SocialLinkProps> = ({ icon, href }) => (
```
в этом случае, например, вместо 3х компонентов можно сделать один, использовать  `enum` для определения значений `Mobile`,`Desktop` и `Ultra`, и при помощи `Condition` возвращать нужные кнопки

oxiqod commented on landing/fragments/landing-library/src/card/card.component.tsx:
```diff
@@ -20,9 +20,9 @@ export const Card: FC<CardKeys> = ({ title, technologiesList, description }) =>
   const { active, activeProps } = useActive()
 
   const getColor = (): string => {
```
вынеси отдельно

oxiqod commented on landing/fragments/landing-library/src/library.component.tsx:
```diff
@@ -22,59 +21,69 @@ export const LibraryBanner = () => {
   const [searchQuery, setSearchQuery] = useState('')
 
   const searchButton = (
```
нужно вынести

oxiqod commented on landing/fragments/landing-library/src/library.component.tsx:
```diff
@@ -22,59 +21,69 @@ export const LibraryBanner = () => {
 
+  const Indent = () => <Layout flexBasis={{ standard: 48, ultra: 72 }} />
+
   const getColor = (): string => {
```
нужно вынести

oxiqod commented on landing/fragments/landing-library/src/library.component.tsx:
```diff
@@ -22,59 +21,69 @@ export const LibraryBanner = () => {
   const [searchQuery, setSearchQuery] = useState('')
```
стрейты тоже можно типизировать

oxiqod commented on landing/fragments/landing-library/src/library-mobile.component.tsx:
какая необходимость делать отдельный компонент для мобильной версии? у тебя для того что бы воплотить всё в одном компоненте есть инструменты: кондишн и брейкпоинты

oxiqod commented on landing/fragments/landing-library/src/card/card.component.tsx:
по логике если компонентом `card` пользуется только компонент `cards`, то и директория должна располагаться в `cards` а не в `src`

oxiqod commented on landing/fragments/landing-navigation/src/index.ts:
```diff
@@ -1,3 +1,5 @@
 export * from './navigation-mobile.component'
 export * from './navigation-index.component'
+export * from './navlink-item'
+export * from './data'
```
шо происходит, куда тебе столько всего? поч для мобилки отдельный компонент? почему `navigation-index.component` не в отдельной директории

если я что то не догнала, то объясни

Mirved64 replied:
> шо происходит, куда тебе столько всего? поч для мобилки отдельный компонент? почему `navigation-index.component` не в отдельной директории
> 
> если я что то не догнала, то объясни

@oksssvv 

Когда делал в одном компоненте, я путался в верстке и решил разделить на мобилку и десктоп (т.к. верстка сильно отличается), а так же на компоненты для главной и других страниц

oxiqod commented on landing/fragments/landing-navigation/src/navlink-item/navlink-item-index.component.tsx:
почему в директории `navlink-item` 4 компонента? тут нужно думать над сменой структуры

![Снимок экрана от 2023-09-23 23-41-08](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/815afefb-3c6b-446e-b72f-fd8d59f10705)

oxiqod commented on landing/fragments/landing-process/src/process.component.tsx:
```diff
@@ -112,11 +112,51 @@ export const Process = () => {
       </Condition>
 
-      <Layout flexBasis={[24, 50]} flexShrink='0' />
+      <Condition match={isWideDesktop || isTV}>
```
color, fontSize, lineHeight и все остальные тоже их принимают брейкпоинты, зачем дублировать текст 

посмотри реализацию этого же компонента: [process](https://github.com/atls-academy/oksssvv.github.io/blob/master/landing/fragments/landing-process/src/process.component.tsx)

oxiqod commented on landing/fragments/landing-steps/src/step/step.component.tsx:
в интерфейсе у тебя для всех пропсов указан `?`, хотя как я посмотрела действительно необязательный только `divider`
на выходе получается вот такой вот интерфейс:
```

export interface StepProps {
  sequenceNumber: string | undefined
  name?: string | undefined
  description?: string | undefined
  divider?: boolean | undefined
}
```

oxiqod commented on landing/fragments/landing-steps/src/steps.component.tsx:
```diff
@@ -70,9 +82,49 @@ export const Steps = () => {
+          </Box>
+        </Condition>
+
+        <Condition match={isTV}>
```
тоже можно переиспользовать, меняется только `width` `height` и брейкпоинт

пример компонента:

```

<Condition match={ breakpoint }>
          <Box
            width={ width }
            height={ height }
            position='absolute'
            zIndex={10}
            margin='auto'
            top='0'
            left='0'
            bottom='0'
            right='0'
          >
            <Image src='./image/Cubics.png' width={ width } height={ height } />
          </Box>
        </Condition>
```

( есть еще другой вид реализации, можешь подсмотреть [Steps](https://github.com/atls-academy/oksssvv.github.io/blob/master/landing/fragments/landing-steps/src/steps.component.tsx#L40-L46) )

oxiqod commented on landing/fragments/landing-steps/src/steps.component.tsx:
```diff
@@ -70,9 +82,49 @@ export const Steps = () => {
+        <Layout flexBasis={160} />
+      </Condition>
+
       <Layout flexBasis={[24, 160]} />
```
зачем кондишн если максимальный лэйаут уже 160? 
поняла, тебе удвоить нужно, дак: `<Layout flexBasis={{_:24, standard:160, ultra:320}} />`

oxiqod commented on landing/fragments/landing-technologies/src/technologies.component.tsx:
```diff
@@ -1,19 +1,27 @@
+        <Layout flexBasis={64} />
+      </Condition>
+
       <Layout flexBasis={[32, 64]} />
```
`<Layout flexBasis={{_:32, standard:64, wide:128, ultra:64}} />` ?

oxiqod commented on landing/pages/library-page/src/library.page.tsx:
```diff
@@ -1,58 +1,249 @@
 
 const LibraryPage = () => {
-  const { isMobile } = useWindowWidth()
+  const { isMobile, isDesktop, isWideDesktop, isTV } = useWindowWidth()
```
ну для начала, зачем повторять слово десктоп в `isWideDesktop`, у тебя ж уже есть подходящие имена для ширин экранов: mobile, desktop, wide и ultra

oxiqod commented on landing/pages/library-page/src/library.page.tsx:
```diff
@@ -1,58 +1,249 @@
-  const { isMobile } = useWindowWidth()
+  const { isMobile, isDesktop, isWideDesktop, isTV } = useWindowWidth()
 
   return (
```
что начинается дальше, я даж не вкурила, дай ссылку на анимацию в фигме, у меня видимо чет глючит, я вижу только скролл до хедера и не более, а уж тем более не вижу анимаций на странице библиотеки

Mirved64 replied:
> что начинается дальше, я даж не вкурила, дай ссылку на анимацию в фигме, у меня видимо чет глючит, я вижу только скролл до хедера и не более, а уж тем более не вижу анимаций на странице библиотеки

@oksssvv 

Анимации и верстка для `wide` и `ultra` на странице библиотеки я сделал сам по аналогии с `index page`. Решил так сделать, чтобы проект был более завершен.

oxiqod commented on landing/fragments/landing-hero/src/social-link/social-link.component.tsx:
```diff
@@ -36,3 +36,18 @@ export const SocialLinkMobile: FC<SocialLinkProps> = ({ icon, href }) => (
+
+export const SocialLinkUltra: FC<SocialLinkProps> = ({ icon, href }) => (
+  <Box>
+    <Link href={href}>
```
сделай открытие ссылок в новой вкладке

oksssvv reviewed (changes requested):
ну и напоследок по внешке:

#### standard
  - слишком рано перекрывается секция курсов
<details>

![Снимок экрана от 2023-09-24 01-03-41](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/b2367512-cb0f-4414-b9eb-9fcc6e026bf1)
</details>

#### wide
  - отступ справа у карт учебного материала и высота самих карточек не 320
  - высота бокса категории
  - секция слишком рано перекрывается следующей
<details>

![Снимок экрана от 2023-09-24 01-07-14](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/4427370f-7a7d-4d11-be88-e52d36cf5ab6)
emy/Mirved64.github.io/assets/126156026/7243dfbe-2eed-45ef-af5d-89ba9ca59253)
</details>

#### ultra
  - драйвер не на всю ширину страницы
  - ширина бокса категорий в драйвере
  - фоновая декорация в степах
  - материалы в library, разве дизайном предусмотрено изменение верстки в библиотеке под широкие экраны? смотрится странно
<details>

![Снимок экрана от 2023-09-24 01-11-07](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/bd8abff5-c380-4c90-8eab-b8f6c9a2d3ae)

![image](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/31a4e805-b9e6-4548-a229-6174e6e404ad)

![Снимок экрана от 2023-09-24 01-26-27](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/2fd1c495-2bf0-4386-bb04-ebd503ea5e93)


</details>

Mirved64 committed:
Merge branch 'feat/landing-fifth-step' into feat/landing-sixth-step

# Conflicts:
#	.pnp.cjs
#	ui/icons/package.json
#	yarn.lock

Mirved64 committed:
chore: check dependencies versions

Mirved64 committed:
refactor: add text components in about section

Mirved64 committed:
chore: rename windowWidth prop isTV to isUltraDesktop

Mirved64 committed:
refactor: divide card component in courses section

Mirved64 committed:
refactor: add text components in info component in courses section

Mirved64 committed:
refactor: rework header component

Mirved64 committed:
refactor: rework hero component

Mirved64 committed:
refactor: rework navigation component

Mirved64 committed:
Merge branch 'feat/landing-fifth-step' into feat/landing-sixth-step

# Conflicts:
#	.pnp.cjs
#	landing/fragments/landing-faq/src/faq.component.tsx
#	landing/fragments/landing-hero/src/hero.component.tsx
#	landing/pages/index-page/src/index.page.tsx
#	landing/pages/library-page/src/library.page.tsx
#	ui/card/src/card.component.tsx
#	ui/slider/src/slider.component.tsx
#	ui/theme/src/theme/backgrounds.ts
#	ui/theme/src/theme/index.ts

Mirved64 committed:
fix: check backgroundColor to backgroundImage
BREAKING CHANGE: backgroundImage not work in courses component and ui/slider

Mirved64 commented:
@oksssvv, @Nelfimov 

**Скоуп**

Отображение градиента как `backgroundImage`

**Описание**

В `ui/slider` не отображается градиент фона слайдера, в компоненте `courses` не отображается градиент фона карточки-ссылки на библиотеку.
Не работают градиенты в `theme/backgrounds.ts` под ключами:

1. `navyGradient`
2. `purpleGradient`
3. `darkPurpleGradient`

в браузере пишет `invalid property value`

**Мои действия**
- гуглил - советуют поставить прификсы к градиенту, но это не помогает;
- Проверял значения градиентов, опечатки в ключах, опечатки в значении пропса `backgroundImage`;
- Если подставить вместо обозначенных выше градиентов такие значения как `navyBlueGradient` или `darkPurpleGradientFlash` - градиенты отображаются;
- Если в браузере вместо пропса `background-image` задать `background`, то градиенты отображается;
- Если хард кодом вставить в пропс `backgroundImage` строку градиента, например `darkPurpleGradient` - градиент не отображается.
- Если хард кодом вставить в пропс `background` строку градиента, например `darkPurpleGradient` - градиент отображается.
- Все остальные градиенты в проекте отображаются как надо, через пропс `backgroundImage` - т.е. не понятна причина почему именно эти три не хотят работать.

**Вопросы** 
Что я мог упустить в этой ситуации? Вроде всё проверил-перепробовал, но причину не обнаружил.
Какие могут быть пути решения?

ПР сделал, ветка актуальна

oxiqod commented:
@Mirved64 
>Отображение градиента как backgroundImage

речь об этих градиентах?
![Снимок экрана от 2023-09-27 17-07-24](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/d1e94d67-88d9-41a4-9dd6-f6ef57781120)

![Снимок экрана от 2023-09-27 17-05-15](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/f81fbc4c-77bc-4731-9391-841bf9e4aca1)

это скрины с запущенного локалхоста, у меня всё отображается

Mirved64 commented:
> @Mirved64
> 
> > Отображение градиента как backgroundImage
> 
> речь об этих градиентах? ![Снимок экрана от 2023-09-27 17-04-55](https://user-images.githubusercontent.com/126156026/271005831-85a7ae19-8b0b-41b7-8a58-2f95ee1cd193.png) ![Снимок экрана от 2023-09-27 17-05-15](https://user-images.githubusercontent.com/126156026/271005876-f81fbc4c-77bc-4731-9391-841bf9e4aca1.png)

@oksssvv 
Я нашел в чем проблема.

В градиенте указан еще цвет бекграунда, т.е. я в проп `backgroundImage` записываю пропсы `backgroundImage` и `backgroundColor`. Разделил проп из фигмы на части и часть с параметром градиента записал в `backgroundImage`, а часть с цветом в `backgroundColor` и всё заработало.

Mirved64 committed:
fix: divide gradient to backgroundImage and backgroundColor

Mirved64 committed:
refactor: move searchButton into search component

Mirved64 committed:
refactor: move getColor function into helpers

Mirved64 committed:
refactor: rework social-link component

Mirved64 committed:
refactor: rework library component and library page

Mirved64 committed:
chore: type useState

Mirved64 committed:
chore: rename windowWidth props

Mirved64 committed:
chore: fix typo in ru.json

Mirved64 committed:
chore: rework social link component

Mirved64 committed:
chore: rework technologies component

Mirved64 committed:
chore: rework steps component

Mirved64 committed:
chore: rework process component

Mirved64 committed:
chore: tune parallax, change backgroundPosition

Mirved64 committed:
chore: fix layout of cards for wide and ultra viewport

Mirved64 committed:
chore: fix layout of header and drawer for ultra viewport

Mirved64 committed:
chore: add text block components into about and courses sections

Mirved64 committed:
feat: add ui/navlinks-block

Mirved64 committed:
refactor: rework header component

Mirved64 committed:
refactor: rework navigation component

Mirved64 committed:
chore: check dependencies in project

Mirved64 committed:
chore: move navigation index component in navigation index directory

oxiqod commented on landing/fragments/landing-courses/src/courses.component.tsx:
мальца слиплось

![Снимок экрана от 2023-10-02 03-21-08](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/22219a79-02dd-4a9c-a8cd-de6f34c66594)

oxiqod replied:
не по макету

![Снимок экрана от 2023-10-02 03-23-49](https://github.com/atls-academy/Mirved64.github.io/assets/126156026/38331acf-bbbc-4978-b268-412122c6e852)

Mirved64 committed:
chore: rework info component in courses section, check typos

Mirved64 committed:
chore: rework header component

Nelfimov commented on landing/fragments/landing-header/src/header.component.tsx:
```diff
@@ -0,0 +1,137 @@
+            <Layout flexBasis={40} />
+
+            <Box display={{ _: 'none', standard: 'flex', ultra: 'none' }}>
+              <Logo fill='rgba(58, 55, 93, 1)' width={56} height={50} />
```
цвета перенести в тему

Nelfimov commented on landing/fragments/landing-header/src/header-index.component.tsx:
```diff
@@ -0,0 +1,162 @@
+            <Layout flexBasis={40} />
+
+            <Box display={{ _: 'none', standard: 'flex', ultra: 'none' }}>
+              <Logo fill='rgba(58, 55, 93, 1)' width={56} height={50} />
```
Цвета в тему

Nelfimov commented on landing/fragments/landing-hero/src/hero.component.tsx:
```diff
@@ -2,154 +2,252 @@ import React                 from 'react'
+import { SocialLinkDisplay } from './social-link'
+
+export const Hero = () => (
+  <Box flexDirection='column' flexGrow='1'>
```
Вынести в отдельный `Column`

Nelfimov commented on landing/fragments/landing-navigation/src/drawer/drawer-card/drawer-card-mobile.tsx:
```diff
@@ -0,0 +1,44 @@
+export const DrawerCardMobile: FC<DrawerCardsProps> = ({ card, index, array }) => (
+  <Column>
+    <Box
+      width={card.category === 'базовый' ? 87 : 122}
```
Хардкод локали

Mirved64 committed:
feat: get color from theme in logo component

Mirved64 committed:
chore: add column wrapper in hero component

Mirved64 committed:
chore: remove hardcode from drawer card mobile

Nelfimov commented on landing/fragments/landing-navigation/src/drawer/drawer-card/drawer-card.interfaces.ts:
```diff
@@ -0,0 +1,17 @@
+export enum DrawerCardCategory { // eslint-disable-line
+  Base = 'базовый',
```
Локаль нельзя хардкодить. Используем только латиницу в коде.

Nelfimov commented on landing/pages/index-page/src/index.page.tsx:
```diff
@@ -1,193 +1,450 @@
+import { useDivHeight }           from '@ui/utils'
 import { useWindowWidth }         from '@ui/utils'
 
 const IndexPage = () => {
```
Выше здоровенный компонент который сложно будет поддерживать. Разбей его на логические части рядом.

Nelfimov commented on landing/pages/library-page/src/library.page.tsx:
```diff
@@ -1,61 +1,235 @@
+import { Box }               from '@ui/layout'
 import { useWindowWidth }    from '@ui/utils'
 
 const LibraryPage = () => {
```
Такая же проблема огромного компонента.

Mirved64 committed:
chore: rework drawer card mobile component

Mirved64 committed:
chore: rework index page, add sections directory

Mirved64 committed:
chore: rework library page, add sections directory

TorinAsakura commented on landing/fragments/landing-about/src/about.component.tsx:
```diff
@@ -1,133 +1,73 @@
+
+          <Space count='6' />
+
+          <Delimiter />
```
ты уже как его только не обозвал, ещё раз - это, долбанный - DIVIDER! https://mui.com/material-ui/react-divider/

Mirved64 replied:
@TorinAsakura 
`Delimiter` в проекте это другой `UI` компонент. 
<details><summary>Details</summary>
<p>

![image](https://github.com/atls-academy/Mirved64.github.io/assets/99605124/81036395-53f7-4f28-ae46-f24010b56ee9)

</p>
</details> 

`Divider` уже есть в `UI`.
<details><summary>Details</summary>
<p>

![image](https://github.com/atls-academy/Mirved64.github.io/assets/99605124/734d0ce6-d274-46bb-a79a-76135e519aad)

</p>
</details>

TorinAsakura commented on landing/fragments/landing-about/src/about.component.tsx:
```diff
@@ -1,133 +1,73 @@
+        </Box>
+
+        <Column display={['flex', 'none']}>
+          <MobileTextBlock id='about.intro-digital-mobile' />
```
если у тебя мобильная сущность настолько ломает гибридное переиспользование компонентов - выноси это в отдельный компонент, если тебе приходится копипастить сущности из-за view - это хреновая практика, возможно я плохо понимаю предназначения этих вещей, но, выглядит как артефакт

Mirved64 replied:
@TorinAsakura 

Здесь я разделил текстовые блоки на мобильный и десктопный, т.к. в мобилке, для соответсвия дизайну, приходится дробить предложение на отдельные фразы, и в мобилке на конце предложения точка, а в десктопе иконка. 

В седьмом степе текстблок сделан под десктоп, т.к. с сервера получаю данные, которые без проблем использую в простом `Text` для мобилки.

TorinAsakura commented on landing/fragments/landing-header/src/header-index.component.tsx:
```diff
@@ -0,0 +1,162 @@
+import { HeaderIndexProps }         from './header.interfaces'
+import { startAnimation }           from './helpers'
+
+export const HeaderIndex: FC<HeaderIndexProps> = ({
```
что за headerIndex?

Mirved64 replied:
@TorinAsakura 

Все компоненты с `Index` в имени предназначены для использования на главной странице. 

Без `Index` в имени для остальных страниц (для странице библиотеки).

В данном случае, хедер с индексом использует линки из библиотеки `react-scroll`, а без индекса линки из `ui/link`

TorinAsakura replied:
вообще не понимаю что за логика

TorinAsakura commented on landing/fragments/landing-library/src/helpers/get-color-icon.ts:
```diff
@@ -0,0 +1,5 @@
+export const getColorIcon = (hover: boolean, active: boolean): string => {
```
это хреновый хелпер, своим неймингом ты декларируешь что он необходим для получения цвета иконки, а сам внутри производишь манипуляции с цветами с проверкой на цвета

либо переименуй либо избавься

TorinAsakura commented on landing/fragments/landing-library/src/helpers/get-color-search-icon.ts:
```diff
@@ -0,0 +1,6 @@
+export const getColorSearchIcon = (hover: boolean, focus: boolean, searchQuery: string): string => {
+  if (hover) return 'darkPurpleTransparent'
+  if (focus) return 'darkestPurple'
+  if (searchQuery) return 'darkestPurple'
```
эти хелперы по своей сути чёрные ящики, то есть, абсолютно не понятно как ими управлять и если полагаться лишь на нейминг то можно обосраться

TorinAsakura commented on landing/fragments/landing-navigation/src/drawer/drawer-card/indent/indent.component.tsx:
```diff
@@ -0,0 +1,13 @@
+
+import { IndentProps } from './indent.interfaces'
+
+export const Indent: FC<IndentProps> = ({ index, array }) => (
```
что это за айндент компоненты? для чего они?? @nelfimov

TorinAsakura commented on landing/fragments/landing-process/src/text-block/index.ts:
```diff
@@ -0,0 +1 @@
+export * from './text-block.component'
```
почему у тебя это везде текстовый блок? а если там будет кнопка? или изображение? побежишь тут же по всей цепи производить переименования?

Mirved64 replied:
@TorinAsakura 

Это попытка в `DRY`, я делю фразу на блоки текста, отличающиеся цветами текста. 
Так код чище выглядит.

TorinAsakura replied:
это неправильный драй который больше отдаёт сломанным солидом)

Mirved64 committed:
chore: remove needless Indent component

Mirved64 committed:
chore: rename helpers in library component

TorinAsakura commented on landing/fragments/landing-library/src/helpers/change-color-icon.ts:
```diff
@@ -1,4 +1,4 @@
-export const getColorIcon = (hover: boolean, active: boolean): string => {
+export const changeColorIcon = (hover: boolean, active: boolean): string => {
```
ты походу не понимаешь…не должно быть зашитой бизнес-логики никакой в названии "поменятьЦветИконки"

Mirved64 committed:
chore: rename helpers

TorinAsakura reviewed (approved):
В общем, с хелперами ты так ничего и не понял походу…

НЕЛЬЗЯ, создавай метод/функцию которая исходя из названия должна "менять цвет" зашивая при этом логику вида: а если у нас серый - то будет бурмалиновый, а, если у нас бурмалиновый то будет серый
иначе
не будет никакого цвета

ты тогда так и называй свои функции, зачем ты людей в заблуждение вводишь?
